cockatrice/src/deck/deck_list_model.cpp

```diff
@@ -632,6 +661,42 @@ QList<QString> *DeckListModel::getZones() const
     return zones;
 }
 
+bool DeckListModel::isCardLegalForCurrentFormat(const CardInfoPtr cardInfo)
+{
+    if (!deckList->getGameFormat().isEmpty()) {
+        if (cardInfo->getProperties().contains("format-" + deckList->getGameFormat())) {
+            return cardInfo->getProperty("format-" + deckList->getGameFormat()) == "legal";
+        } else {
+            return false;
+        }
+    }
+    return true;
+}
+
+void DeckListModel::refreshCardFormatLegalities()
+{
+    InnerDecklistNode *listRoot = deckList->getRoot();
+
+    for (int i = 0; i < listRoot->size(); i++) {
+        auto *currentZone = dynamic_cast<InnerDecklistNode *>(listRoot->at(i));
```

Comment:
why dynamic cast this if you're not going to nullcheck it? might as well use a static cast right?